Added simple base for a `has_any_role` decorator check.

patchwork-systems opened the pull request:
I just wanted to submit this for an initial review. I'd be happy to make any changes that might be necessary!

Commit:
Added simple base for a `has_any_role` decorator check.

nxtlo reviewed (changes requested):
Ain't this suppose to check if the member was human? or `None`?

Commit:
Changes based on initial PR feedback.

- with_any_roles_check internal logic separated into classes for compatibility
- Removed 3.10 only features
